### PR TITLE
add slither github action

### PR DESCRIPTION
**Pull request by gfournieriExec**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -21,3 +21,18 @@ jobs:
...
           slug: iExecBlockchainComputing/iexec-voucher-contracts
+      - name: Run static analysis with slither
+        uses: crytic/slither-action@v0.4.0
+        id: slither
+        with:
+          target: "contracts/"
+          solc-version: '0.8.21'
+          slither-args: --checklist --markdown-root ${{ github.server_url }}/${{ github.repository }}/blob/${{ github.sha }}/
+          node-version: "20"
+          fail-on: none
```
Comment (jeremyjams) [inline .github/workflows/main.yml:32]: Is the build because of this line?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -21,3 +21,18 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           slug: iExecBlockchainComputing/iexec-voucher-contracts
+      - name: Run static analysis with slither
+        uses: crytic/slither-action@v0.4.0
+        id: slither
+        with:
+          target: "contracts/"
+          solc-version: '0.8.21'
```
Comment (jeremyjams) [inline .github/workflows/main.yml:29]: Do we really need this?

File: hardhat.config.ts [hunk 3]
```diff
@@ -62,6 +62,7 @@ const config: HardhatUserConfig = {
     },
     dependencyCompiler: {
         paths: ['@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol'],
+        keep: true,
```
Comment (jeremyjams) [inline hardhat.config.ts:65]: Maybe add a note that it has been introduced for slither

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -21,3 +21,18 @@ jobs:
...
+          slither-args: --checklist --markdown-root ${{ github.server_url }}/${{ github.repository }}/blob/${{ github.sha }}/
+          node-version: "20"
+          fail-on: none
+          sarif: results.sarif
+          slither-config: 'slither.config.json'
```
Comment (jeremyjams) [inline .github/workflows/main.yml:34]:
```suggestion
```
Already the default

> The path to the Slither configuration file. By default, ./slither.config.json is used if present. See [Configuration file](https://github.com/crytic/slither/wiki/Usage#configuration-file).

File: .github/workflows/main.yml (hunk 1 above)
Comment (gfournieriExec) [reply .github/workflows/main.yml:32]: Yes

**Review by jeremyjams — APPROVED** [review]
Very nice thanks! We'll try to remove `fail-on: none` asap :)

File: .github/workflows/main.yml (hunk 1 above)
Comment (zguesmi) [reply .github/workflows/main.yml:31]: Can we mark it as TODO to not forget?

**Review by zguesmi — APPROVED** [review]
Well done!

**Pull request merged**